tests/behat/behat-features/src/JsonApiContext.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+/**
+ * @file
+ * The main behat context.
+ */
+
+use Behat\Mink\Exception\ExpectationException;
+use GuzzleHttp\Psr7\Request;
+use Http\Adapter\Guzzle6\Client;
+use WoohooLabs\Yang\JsonApi\Request\JsonApiRequestBuilder;
+use WoohooLabs\Yang\JsonApi\Client\JsonApiClient;
+use WoohooLabs\Yang\JsonApi\Hydrator\ClassHydrator;
+use Drupal\DrupalExtension\Context\RawDrupalContext;
+
+/**
+ * Defines application features from the specific context.
+ */
+class JsonApiContext extends RawDrupalContext {
+
+  /**
+   * Satellite base url.
+   *
+   * @var string
+   */
+  protected $satelliteBaseUrl;
+
+  /**
+   * Contentpool base url.
+   *
+   * @var string
+   */
+  protected $contentpoolBaseUrl;
+
+  /**
+   * Consumer id.
+   *
+   * @var string
+   */
+  protected $consumerId;
+
+  /**
+   * Json api client.
+   *
+   * @var WoohooLabs\Yang\JsonApi\Client\JsonApiClient
+   */
+  protected $jsonApiClient;
+
+  /**
+   * @BeforeScenario
+   */
+  public function before($scope) {
+    if (!getenv('PHAPP_BASE_URL')) {
+      throw new Exception('Missing satellite base URL.');
+    }
+    $this->satelliteBaseUrl = getenv('PHAPP_BASE_URL');
+    if (!getenv('CONTENTPOOL_BASE_URL')) {
+      throw new Exception('Missing contentpool base URL.');
+    }
+    $this->contentpoolBaseUrl = getenv('CONTENTPOOL_BASE_URL');
+
+    $client = Client::createWithConfig([]);
+
+    $this->jsonApiClient = new JsonApiClient($client);
+  }
+
+  /**
+   * @Then I request an article with the uuid :uuid
+   */
+  public function iRequestAnArticleWithTheUuid($uuid) {
+    // Instantiate an empty PSR-7 request.
+    $request = new Request("", "");
+
+    $requestBuilder = new JsonApiRequestBuilder($request);
+
+    $requestBuilder
+      ->setProtocolVersion("1.1")
+      ->setMethod("GET")
+      ->setUri("{$this->contentpoolBaseUrl}/jsonapi/node/article/$uuid")
+      ->setHeader("Accept-Charset", "utf-8");
+
+    $request = $requestBuilder->getRequest();
+
+    $response = $this->jsonApiClient->sendRequest($request);
+
+    if (!$response->isSuccessful() || !$response->hasDocument()) {
+      throw new ExpectationException('Request to fetch an article was not successful or has no results', $this->getSession());
+    }
+
+  }
+
+  /**
+   * @Then I save the consumer id
```

Comment:
would be nice to provide more context, e.g. "Then I save the consumer id for oauth authentication"